widgetssdk/src/test/java/com/glia/widgets/filepreview/data/GliaFileRepositoryTest.java

```diff
@@ -0,0 +1,149 @@
+package com.glia.widgets.filepreview.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import android.graphics.Bitmap;
+
+import com.glia.androidsdk.GliaException;
+import com.glia.androidsdk.RequestCallback;
+import com.glia.androidsdk.chat.AttachmentFile;
+import com.glia.widgets.chat.helper.FileHelper;
+import com.glia.widgets.di.GliaCore;
+import com.glia.widgets.filepreview.data.source.local.DownloadsFolderDataSource;
+import com.glia.widgets.filepreview.data.source.local.InAppBitmapCache;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.InputStream;
+
+import io.reactivex.Completable;
+import io.reactivex.Maybe;
+
+public class GliaFileRepositoryTest {
+
+    private GliaFileRepository gliaFileRepository;
+    private InAppBitmapCache bitmapCache;
+    private DownloadsFolderDataSource downloadsFolderDataSource;
+    private GliaCore gliaCore;
+    private FileHelper fileHelper;
+
+    @Before
+    public void setUp() {
+        bitmapCache = mock(InAppBitmapCache.class);
+        downloadsFolderDataSource = mock(DownloadsFolderDataSource.class);
+        gliaCore = mock(GliaCore.class);
+        fileHelper = mock(FileHelper.class);
+        gliaFileRepository = new GliaFileRepositoryImpl(
+                bitmapCache,
+                downloadsFolderDataSource,
+                gliaCore,
+                fileHelper
+        );
+    }
+
+    @Test
+    public void loadImageFromCache_returnsMaybeOnSuccess() {
+        when(bitmapCache.getBitmapById(FILENAME)).thenReturn(BITMAP);
+        Maybe<Bitmap> result = gliaFileRepository.loadImageFromCache(FILENAME);
+        result.test().assertResult(BITMAP);
+    }
+
+    @Test
+    public void loadImageFromCache_returnsMaybeOnError() {
+        doThrow(RUNTIME_EXCEPTION).when(bitmapCache).getBitmapById(FILENAME);
+        Maybe<Bitmap> result = gliaFileRepository.loadImageFromCache(FILENAME);
+        result.test().assertError(RUNTIME_EXCEPTION);
+    }
```

Comment:
Should also test a case when Bitmap returned by the `bitmapCache.getBitmapById` was `null`